tsl/test/src/test_compression.c

```diff
@@ -319,42 +319,117 @@ test_gorilla_float()
 	TestAssertInt64Eq(i, 0);
 }
 
+static uint64
+test_hash64(uint64 x)
+{
+	x ^= x >> 30;
+	x *= 0xbf58476d1ce4e5b9U;
+	x ^= x >> 27;
+	x *= 0x94d049bb133111ebU;
+	x ^= x >> 31;
+	return x;
+}
+
 static void
-test_gorilla_double()
+test_gorilla_double(bool have_nulls, bool have_random)
 {
 	GorillaCompressor *compressor = gorilla_compressor_alloc();
 	GorillaCompressed *compressed;
-	DecompressionIterator *iter;
-	double i;
-	for (i = 0.0; i < 1015.0; i++)
-		gorilla_compressor_append_value(compressor, double_get_bits(i));
+
+	double values[1015];
+	bool nulls[1015];
+	for (int i = 0; i < 1015; i++)
+	{
+		if (have_random)
+		{
+			/* Also add some stretches of equal numbers. */
+			int base = i;
+			if (i % 37 < 3)
+			{
+				base = 1;
+			}
+			else if (i % 53 < 2)
+			{
+				base = 2;
+			}
+
+			values[i] = (test_hash64(base) / (double) PG_UINT64_MAX) * 100.;
+		}
+		else
+		{
+			values[i] = i;
+		}
+
+		if (have_nulls && i % 29 == 0)
+		{
+			nulls[i] = true;
+		}
+		else
+		{
+			nulls[i] = false;
+		}
+
+		if (nulls[i])
+		{
+			gorilla_compressor_append_null(compressor);
+		}
+		else
+		{
+			gorilla_compressor_append_value(compressor, double_get_bits(values[i]));
+		}
+	}
 
 	compressed = gorilla_compressor_finish(compressor);
 	TestAssertTrue(compressed != NULL);
-	TestAssertInt64Eq(VARSIZE(compressed), 1200);
+	if (!have_nulls && !have_random)
+	{
+		TestAssertInt64Eq(VARSIZE(compressed), 1200);
+	}
 
-	i = 0;
-	iter =
+	/* Forward decompression. */
+	DecompressionIterator *iter =
 		gorilla_decompression_iterator_from_datum_forward(PointerGetDatum(compressed), FLOAT8OID);
-	for (DecompressResult r = gorilla_decompression_iterator_try_next_forward(iter); !r.is_done;
-		 r = gorilla_decompression_iterator_try_next_forward(iter))
+	ArrowArray *bulk_result =
```

Comment:
Are you missing an include here?

Reply:
Oh, I missed half the files I wanted to include, will fix now.